weave/trace_server/file_storage_credentials.py

```diff
@@ -0,0 +1,115 @@
+import base64
+from typing import Optional, TypedDict, Union
+
+from azure.core.credentials import TokenCredential
+from google.oauth2.credentials import Credentials as GCPCredentials
+
+from weave.trace_server import environment
+
+
+class AWSCredentials(TypedDict, total=False):
+    """AWS authentication credentials for S3 access.
+    The session_token is optional and only required for temporary credentials."""
+
+    access_key_id: str
+    secret_access_key: str
+    session_token: Optional[str]
+
+
+class AzureConnectionCredentials(TypedDict):
+    """Azure authentication using connection string format."""
+
+    connection_string: str
+
+
+class AzureAccountCredentials(TypedDict):
+    """Azure authentication using account-based credentials."""
+
+    credential: Union[str, TokenCredential]
+    account_url: Optional[str]
+
+
+def get_aws_credentials() -> AWSCredentials:
+    """Retrieves AWS credentials from environment variables.
+
+    Required env vars:
+        - WF_FILE_STORAGE_BUCKET_AWS_ACCESS_KEY_ID
+        - WF_FILE_STORAGE_BUCKET_AWS_SECRET_ACCESS_KEY
+    Optional env vars:
+        - WF_FILE_STORAGE_BUCKET_AWS_SESSION_TOKEN
+
+    Returns:
+        AWSCredentials with access key, secret key, and optional session token
+
+    Raises:
+        ValueError: If required credentials are not set
+    """
+    access_key_id = environment.wf_storage_bucket_aws_access_key_id()
+    secret_access_key = environment.wf_storage_bucket_aws_secret_access_key()
+    session_token = environment.wf_storage_bucket_aws_session_token()
+    if access_key_id is None or secret_access_key is None:
+        raise ValueError("AWS credentials not set")
+
+    creds: AWSCredentials = {
```

Comment:
Safer to use the TypedDict directly here

Reply:
done